propolis/Cargo.toml

```diff
@@ -21,7 +21,8 @@ viona_api = { path = "../viona-api" }
 usdt = { version = "0.3.2", default-features = false }
 tokio = { version = "1", features = ["full"] }
 futures = "0.3"
-crucible = { git = "https://github.com/oxidecomputer/crucible", rev = "fed3e8ca7762130ee146fc516a4ef6eed2b91629", optional = true }
+crucible = { git = "https://github.com/oxidecomputer/crucible", branch = "producemetric", optional = true }
+oximeter = { git = "https://github.com/oxidecomputer/omicron", branch = "main", optional = true }
```

Comment:
Just to make sure I'm not misreading: it looks like `oximeter` is only optional if `crucible` is disabled; otherwise the Crucible module will be missing the `ProducerRegistry` type. Is that right? (I'm trying to reason through how we might keep this module optional when we extend metrics collection to more than just Crucible.)

Reply:
I'm actually not 100% sure, but my intent here was trying to allow propolis standalone to build and not have to pull in both crucible and then oximeter.

Crucible needs to know about ProducerRegistry, and now propolis-server will also need to know about it, even if Crucible is not present.